app/controllers/json_api/people_controller.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2022, Schweizer Wanderwege. This file is part of
+#  hitobito and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito.
+
+class JsonApi::PeopleController < JsonApiController
+
+  def index
+    authorize!(:index_people, Group)
+    people = PersonResource.all(params, index_people_scope)
```

Comment:
Ist es nicht eine Sicherheitslücke, wenn die `params` komplett reingegeben werden? Ich habe das Gefühl, dass hier `strong_params` sinnvoll wäre.

Reply:
nein, das ist laut graphiti gem doc ok so weil die Resource smart ist und dies handelt.
betrifft ja nur schreibende operationen.
aber vertrauen ist gut, kontrolle ist besser. 
habe noch zwei specs ergänzt die zeigen das ein 400er zurück gegeben wird falls da jemand was krumes versuchen will. 

https://github.com/hitobito/hitobito/blob/feature/json-api/spec/controllers/json_api/people_controller_spec.rb#L973 sowie https://github.com/hitobito/hitobito/blob/feature/json-api/spec/controllers/json_api/people_controller_spec.rb#L994

Reply:
es ist wichtig das wir das genau anschauen und uns einig sind!
* habe gerade noch geprüft wie das in einem anderen Rails Projekt gemacht wird das Graphiti verwendet. Genau gleich wie wir jetzt hier, die params werden direkt der Resourcen Klasse übergeben. (klar NICHT dem model direkt) [siehe](https://gitlab.puzzle.ch/bafu/sam/-/blob/master/core/app/controllers/graphiti_controller.rb)
* In der Graphiti rails example app wird es auch so gemacht: https://github.com/graphiti-api/employee_directory/blob/master/app/controllers/departments_controller.rb#L3
* [Test Case](https://github.com/hitobito/hitobito/blob/feature/json-api/spec/controllers/json_api/people_controller_spec.rb#L973), sowie [Test Case](https://github.com/hitobito/hitobito/blob/feature/json-api/spec/controllers/json_api/people_controller_spec.rb#L994) laufen durch und zeigen das Graphiti Resource diese cases handelt.
* `find` und `all` sind nicht die methoden der model klassen, sondern der zugehörigen  Graphiti Resourcen

hier noch aus der Doku/Source von Graphiti wie entries erstellt/updated werden:

```ruby
def create(attributes)
  employee = Employee.new
  attributes.each_pair do |key, value|
    employee.send(:"#{key}=", value)
  end
  employee.save
  employee
end

def update(attributes)
  employee = EmployeeResource.find(attributes.delete(:id)).data
  attributes.each_pair do |key, value|
    employee.send(:"#{key}=", value)
  end
  employee.save
  employee
end

def destroy(attributes)
  employee = EmployeeResource.find(attributes.delete(:id)).data
  employee.destroy
  employee
end
```

@carlobeltrame habe ich noch was übersehen? willst du nochmals drüber schauen?

Reply:
Ah, jetzt hats glaube ich geklickt bei mir. Du sagst also, die Resource gibt die Params nicht einfach weiter ans Model, sondern schaut natürlich in der Resourcen-Definition nach, welche Filter erlaubt sind, welche Felder schreibbar sind, etc.. Macht Sinn.
Dein sad path Test Case ist aber immer noch falsch, auf https://github.com/hitobito/hitobito/blob/feature/json-api/spec/controllers/json_api/people_controller_spec.rb#L999 befüllst du das ID-Feld im Payload mit dem Namen von inaccessible_person statt mit der ID (to_s gibt den Namen, nicht die ID).

Reply:
ah damn, sehr gut das du da gesehen hast. 
so scheint es wirklich möglich eine Person zu ändern auf die man keinen Zugriff hat.
ich passe den controller noch entsprechend an.

Reply:
```
  def self.find(params = {}, base_scope = nil)
    # make sure both id params are the same
    # for update since we're checking permission based on
    # params :id
    data_id = params[:data].try(:[], :id).try(:to_i)
    param_id = params[:id].to_i
    if data_id && param_id
      raise ActionController::BadRequest unless data_id == param_id
    end

    super(params, base_scope)
  end
```

habe gleich die find Methode der ApplicationResource überschrieben. So stellen wir sicher das es gleich für künftige Resourcen auch implementiert ist.

Reply:
Ist das in dem Fall nicht ein generelles Problem mit Graphiti, was wir dort melden müssten? Dann könnten sie allenfalls ihre Dokumentation anpassen damit klar wird, welches Objekt / welche ID man auf Zugriff überprüfen müsste. Oder müssten wir wenigstens Andreas noch benachrichtigen?

Reply:
finde ich gut. ich informiere Andreas gerne und mache noch ein Issue bei Graphiti.

Reply:
https://github.com/graphiti-api/graphiti/issues/441
habe den link auch an andreas geschickt.